.github/workflows/cigh.yml

```diff
@@ -0,0 +1,124 @@
+# This workflow aims at building and testing FreeCAD
+# It uses a runner based on latest Ubuntu LTS version
+# It builds using CMake/CCache+Make/Clang as a toolchain
+#
+# It has been foreseen to be used with CMake/Ninja/Clang...
+# ...but this is disabled at the moment because Ninja...
+# ...only works based on file timestamps, which isn't...
+# ...compatible with Git not managing them
+
+name: FCBuildNTest
+
+on:
+  workflow_dispatch: # Triggered manually
+# To run manually, use :
+# curl -X POST -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/:OWNER/FreeCAD/actions/workflows/cigh.yml/dispatches -d '{"ref":":BRANCH_NAME"}' -u :USER_NAME
+# It will then require your Github PAT to trigger the launching event
+    branches-ignore: # Do not run on master and release branches
+      - master
+      - releases/**
+  pull_request:
+    branches-ignore: # Do not run on master and release branches
+      - master
+      - releases/**
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  BuildNTest:
+    runs-on: ubuntu-latest
+    container: openbrain/freecad
```

Comment:
I suggest to put the Dockerfile used to create this image somewhere in this repo, too. Or place at least a comment on where to find it. This would make it easier for other people to update/maintain and keep it alive.

Reply:
That's foreseen. It's here ATM : https://github.com/0penBrain/CItest/blob/master/Dockerfile
But most probably if something happens with this, it's better to use an image stored in some FC global Docker hub.